### ISSUE TITLE
Bad InfiniteScroll path option. Next link not found: .pagination__next

### ISSUE THREAD

**Issue by neorganic** [metafizzy/infinite-scroll#707]
> Bad InfiniteScroll path option. Next link not found: .pagination__next
> Disabling InfiniteScroll
> Uncaught TypeError: this.getPath is not a function

When next link is not present(it happens if have only 1 record), the script brakes, braking all my other js scripts. How could I resolve this problem without needing to every time have the next link present ?

**Comment by desandro** [comment]
Thanks for reporting this issue. This bug was fixed in v3.0.1. See #691 . Can you double check you are using the latest version?

**Comment by neorganic** [comment]
Yeah, I was using  v3.0.0  . After moving to v3.0.1 the issue seems to be resolved. Great work!

**Comment by kavin-90** [comment]
I'm facing the same issue on 3.0.2 

can u please check?

**Comment by filtah** [comment]
have the same issue on 3.0.2. 
3.0.1 is causing the same!
I'll check my code..

**Comment by filtah** [comment]
well not sure if its meant to work like that, but go rid of the error by checking for the element before init of infinite scroll.

if ($('#moreProperties').length) {
  //init here..
}

**Comment by sharifzadesina** [comment]
@desandro This is also not fixed.

**Comment by desandro** [comment]
I've re-opened the issue. I'll be taking another look at this.

**Comment by sharifzadesina** [comment]
@desandro Thank you.
When you want render a page you write is like this:
```
for article in articles:
    render article
if we have more pages:
   render pagination
```
We don't render pagination always, but when we do it right this we get error if we don't have more pages.

**Comment by desandro** [comment]
This is a duplicate of #691. I'm copying my answer from there, and closing this issue here

---

To be clear, The `Uncaught TypeError: this.getPath is not a function` JS error should be fixed as of v3.0.2. The other error messages still occur.

As of v3.0.2 the intended behavior is on last pages where Infinite Scroll is initialized and `path` is set to a selector string, and that selected element is not available; Infinite Scroll will log two error messages:

```
Bad InfiniteScroll path option. Next link not found: .pagination__next
Disabling InfiniteScroll
```

These are error messages and not thrown JS errors, so they will not disrupt JS. These messages are useful for users who have set `path` to the wrong element, or are trying to use Infinite Scroll on a page with out a next link in the HTML.
